### PR TITLE
Fix mapping logic

### PR DESCRIPTION
**Pull request by Golo300**

### CODE REVIEW

File: lib/side.py [hunk 1]
```diff
@@ -33,26 +39,31 @@ def transfer_up(self, pos):
...
-                pos.x = self.size - 1                
+                pos.x = self.size - 1    
+
+            return self.invert(self.top_dir)           
 
     def transfer_down(self, pos):
         if hasattr(self, "down"):
             pos.side = self.down
             pos.y = self.size - 1
+            return self.invert(self.down_dir)
```
Comment (rubenhoenle) [inline lib/side.py:50]: no return statement if `hasattr(self, "down")` is False

File: lib/side.py [hunk 2]
```diff
@@ -33,26 +39,31 @@ def transfer_up(self, pos):
...
             elif self.left_dir == Direction.UP:
-                pos.x = pos.y
+                pos.x = self.size - pos.y
                 pos.y = self.size - 1
             elif self.left_dir == Direction.DOWN:
                 pos.x = self.size - pos.y - 1
                 pos.y = 0
 
-            pos.x = self.size - 1
+            #pos.x = self.size - 1
```
Comment (rubenhoenle) [inline lib/side.py:64]: is that commented out code still needed

File: lib/side.py [hunk 3]
```diff
@@ -66,6 +77,18 @@ def transfer_right(self, pos):
                 pos.x = pos.y
                 pos.y = 0
 
+            return self.invert(self.right_dir)
```
Comment (rubenhoenle) [inline lib/side.py:80]: no return statement if `hasattr(self, "right")` is False

**Review by rubenhoenle — CHANGES_REQUESTED** [review]
see above

File: lib/side.py (hunk 1 above)
Comment (Golo300) [reply lib/side.py:51]: fixed by raising exception

File: lib/side.py (hunk 3 above)
Comment (Golo300) [reply lib/side.py:81]: see above

**Review by rubenhoenle — APPROVED** [review]

**Pull request merged**